### PR TITLE
feat(ffi): screenshot example using C# bindings

### PR DESCRIPTION
**Pull request by irvingoujAtDevolution**
1. upgraded the connect example to screenshot example.
2. exposed the active stage interface
3. upgraded and moved the utility classes on the C# side.

### CODE REVIEW

File: ffi/dotnet/Devolutions.IronRdp.ConnectExample/Devolutions.IronRdp.ConnectExample.csproj [hunk 1]
```diff
@@ -10,5 +10,10 @@
   <ItemGroup>
     <ProjectReference Include="../Devolutions.IronRdp/Devolutions.IronRdp.csproj" />
   </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Windows.Compatibility" Version="8.0.3" />
+    <PackageReference Include="SixLabors.ImageSharp" Version="3.1.3" />
```
Comment (CBenoit) [inline ffi/dotnet/Devolutions.IronRdp.ConnectExample/Devolutions.IronRdp.ConnectExample.csproj:16]: question: Is it a leftover? If not, can look into using the standard library instead?

There are quite a few primitives available even in .NET core:
- https://learn.microsoft.com/en-us/dotnet/api/system.drawing?view=dotnet-plat-ext-8.0
- https://learn.microsoft.com/en-us/dotnet/api/system.drawing.image?view=dotnet-plat-ext-8.0
- https://learn.microsoft.com/en-us/dotnet/api/system.drawing.bitmap?view=dotnet-plat-ext-8.0

**Review by CBenoit — COMMENTED** [review]
Good job!

File: ffi/dotnet/Devolutions.IronRdp.ConnectExample/Devolutions.IronRdp.ConnectExample.csproj [hunk 2]
```diff
@@ -10,5 +10,10 @@
   <ItemGroup>
     <ProjectReference Include="../Devolutions.IronRdp/Devolutions.IronRdp.csproj" />
   </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Windows.Compatibility" Version="8.0.3" />
```
[the remark this answers is not in the document]
Comment (irvingoujAtDevolution) [reply ffi/dotnet/Devolutions.IronRdp.ConnectExample/Devolutions.IronRdp.ConnectExample.csproj]: updated, I was trying to create a working client example,

File: ffi/src/input.rs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#[diplomat::bridge]
+pub mod ffi {
+    #[diplomat::opaque]
+    pub struct Database(pub ironrdp::input::Database);
```
[the remark this answers is not in the document]
Comment (irvingoujAtDevolution) [reply ffi/src/input.rs]: updated

File: ffi/src/connector/mod.rs [hunk 4]
```diff
@@ -132,9 +132,9 @@ pub mod ffi {
     }
 
     #[diplomat::opaque]
-    pub struct State<'a>(pub &'a dyn ironrdp::connector::State);
+    pub struct DynClientConnectorState<'a>(pub &'a dyn ironrdp::connector::State);
```
[the remark this answers is not in the document]
Comment (irvingoujAtDevolution) [reply ffi/src/connector/mod.rs]: updated

File: ffi/src/utils/mod.rs [hunk 5]
```diff
@@ -28,6 +28,56 @@ pub mod ffi {
...
+        pub fn fill(&'a self, buffer: &'a mut [u32]) -> Result<(), Box<IronRdpError>> {
+            if buffer.len() < self.0.len() {
+                return Err("Buffer is too small".into());
+            }
+            buffer.copy_from_slice(self.0);
+            Ok(())
+        }
+    }
+
+    #[diplomat::opaque]
```
[the remark this answers is not in the document]
Comment (irvingoujAtDevolution) [reply ffi/src/utils/mod.rs]: updated

File: ffi/dotnet/Devolutions.IronRdp.ConnectExample/Program.cs [hunk 6]
```diff
@@ -21,15 +21,95 @@ static async Task Main(string[] args)
...
+                        var pduReadTask = await readPduTask;
+                        action = pduReadTask.Item1;
+                        payload = pduReadTask.Item2;
+                        Console.WriteLine($"Action: {action}");
+                    }
+                    else
+                    {
+                        Console.WriteLine("Timeout");
+                        keepLooping = false;
+                        continue;
```
[the remark this answers is not in the document]
Comment (irvingoujAtDevolution) [reply ffi/dotnet/Devolutions.IronRdp.ConnectExample/Program.cs]: updated

File: ffi/dotnet/Devolutions.IronRdp.ConnectExample/Devolutions.IronRdp.ConnectExample.csproj (hunk 1 above)
Comment (irvingoujAtDevolution) [reply ffi/dotnet/Devolutions.IronRdp.ConnectExample/Devolutions.IronRdp.ConnectExample.csproj]: updated

File: ffi/dotnet/Devolutions.IronRdp.ConnectExample/Devolutions.IronRdp.ConnectExample.csproj (hunk 2 above)
[the remark this answers is not in the document]
Comment (CBenoit) [reply ffi/dotnet/Devolutions.IronRdp.ConnectExample/Devolutions.IronRdp.ConnectExample.csproj]: No problem! Try to clean up before marking the PR as ready

**Review by CBenoit — APPROVED** [review]
Thank you!

**Pull request merged**